### DISCUSSION TITLE
Attribute #[Validate(properties: [], after: true)]

### DISCUSSION THREAD

**Discussion by MrPunyapal** [livewire/livewire#7129] [3. Feature ideas]
what if we can have new attribute called validate and we can just define it above the method and it validates given properties before or after calling method?

for example we have count component and count property with method increment and decrement and we want to validate the count property after calling the method the ensure that if it met the rules.

therefore we can define attribute like this 👇

```php
  #[Rule('numeric|min:0|max:10')]
  public $count = 1;

  #[Validate(properties: ['count'], after: true)]
  public function increment()
  {
      $this->count++;
  }
```

I have also tried to create attribute it would be look like this? 🤔

```php
use Livewire\Features\SupportAttributes\Attribute as LivewireAttribute;
use Attribute;
use function Livewire\store;

#[Attribute(Attribute::IS_REPEATABLE | Attribute::TARGET_METHOD)]
class BaseValidate extends LivewireAttribute
{
    function __construct(public string|array|null $properties = null, public bool $after = false)
    {
    }

    function boot()
    {
        $properties = [];

        if (is_array($this->properties) && count($this->properties) > 0) {
            $properties = $this->properties;
        } else {
            $properties[] = $this->properties;
        }

        store($this->component)->push(
            $this->after ? 'validationPropertiesForMethodAfter' : 'validationPropertiesForMethod',
            $properties,
            $this->getName(),
        );
    }
}
```

and I have also tried to do some validation before calling the method something like this and tried to put into any of component hook but was getting some errors or I don't know where I have to place them.

```php
$validationPropertiesForMethod = store($this->component)->get('validationPropertiesForMethod');

if (key_exists($method, $validationPropertiesForMethod)) {
    $properties = $validationPropertiesForMethod[$method];
    $this->component->validate(blank($properties)?null:collect($this->component->getRules())
    ->filter(function ($value, $key) use ($properties) {
        return in_array(str($key)->before('.*'), $properties);
    })
    ->toArray());
}
```

error I was facing is `Method Livewire\Features\SupportRedirects\Redirector::withInput does not exist.`

![image](https://github.com/livewire/livewire/assets/53343069/de41d449-2237-4b90-ae7b-790716853d7f)

I hope it can help

**Comment by lrljoe** [comment]
This already exists in v3: [https://livewire.laravel.com/docs/validation](https://livewire.laravel.com/docs/validation)

Confused as to what you're suggesting is added here?

**Reply by MrPunyapal** [reply]
It's not a attribute 😅

**Comment by joshhanley** [comment]
https://github.com/livewire/livewire/pull/7260 👀

**Reply by MrPunyapal** [reply]
That PR merged now 😅
